Reemit.Gui/ViewModels/HelloViewModel.cs

```diff
@@ -1,12 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Avalonia.Platform.Storage;
 using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+using Reemit.Decompiler;
 
 namespace Reemit.Gui.ViewModels;
 
 public class HelloViewModel : ReactiveObject, IRoutableViewModel
 {
+    public ReactiveCommand<Unit, Unit> OpenFiles { get; }
+
+    public ReactiveCommand<Unit, Unit> OpenDroppedFiles { get; }
+
+    [Reactive]
+    public IReadOnlyList<IStorageItem>? FilesDraggedOver { get; set; }
+
+    [ObservableAsProperty]
+    public bool IsAcceptedFileDraggedOver { get; }
+
     public string UrlPathSegment => nameof(HelloViewModel);
     
     public IScreen HostScreen { get; }
 
-    public HelloViewModel(IScreen screen) => HostScreen = screen;
+    public Interaction<Unit, IReadOnlyList<IStorageItem>> ShowOpenFileDialog { get; } = new();
+
+    public HelloViewModel(IScreen hostScreen)
+    {
+        HostScreen = hostScreen;
+
+        this.WhenAnyValue(vm => vm.FilesDraggedOver)
+            .Select(ShouldAcceptDraggedFiles)
+            .ToPropertyEx(this, vm => vm.IsAcceptedFileDraggedOver);
+
+        OpenFiles = ReactiveCommand.CreateFromTask<Unit, Unit>(OpenFilesAsync);
+
+        var hasAnyDraggedOverFiles = this.WhenAnyValue(vm => vm.FilesDraggedOver)
+            .Select(x => x is { Count: > 0 });
```

Comment:
Tested on Windows, it seems this is triggered despite `ShouldAcceptDraggedFiles` returning `false`:

![image](https://github.com/v0idzz/Reemit/assets/7999934/60a10e08-51e9-43bf-b973-71d4b57d679a)

Not sure if I'm using the right terminology here, but there seems to be something of a race condition, with `OpenDroppedFilesAsync` sometimes being invoked:

![image](https://github.com/v0idzz/Reemit/assets/7999934/1f07c3fa-fbe0-4e93-afa8-de17fb11268a)

Not sure how "real world" this bug is as it does not seem to repro without breakpoints being set, but the fact that it was observed under a debugger makes me suspect it may possibly occur in real world scenarios. Is there a different way to handle this to ensure the validation is always performed? Or am I simply hitting a known quirk that is a side effect of having a debugger attached?

Reply:
Woopsie. Was using wrong property there. Please verify if this still happens--it should not.

Reply:
All appears well.